CMakeLists.txt

```diff
@@ -82,6 +82,12 @@ if(CMAKE_C_FLAGS_RELEASE MATCHES "-O3")
   string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
 endif()
 
+# Enable link-time optimisations on release builds.
+check_c_compiler_flag(-flto HAS_FLTO_FLAG)
+if(HAS_FLTO_FLAG)
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -flto")
```

Comment:
Confirmed, only adding to `CMAKE_C_FLAGS` leads to errors like this in the Release build:

```
[103/108] Building C object src/nvim/CMakeFiles/nvim.dir/regexp.c.o
clang: warning: -lfto: 'linker' input unused
```